File: src/assets/download_images.js
const https = require("https");
const fs = require("fs");
const { exit } = require("process");
const Stream = require("stream").Transform

let rawdata = fs.readFileSync("achievements.json");
let achievements = JSON.parse(rawdata);

achievements.forEach(async (achievement) => {
  let filename = "images/" + achievement.identifier + ".png"
  fs.stat(filename, function (err, stats) {
    if (err) {
      console.log("new image", achievement.identifier)
    } else {
      if (stats.size < 2000) {
        console.log("image", filename, "is small and likely corrupted. Will be overwritten")
      } else {
        return
      }
    }
  });
  https.get(achievement.url, (res) => {
    var data = new Stream();
    res.on('data', chunk => {
        data.push(chunk);
    });
    res.on('end', () => {
        fs.writeFileSync(filename, data.read());
    })
  });
});

console.log('We done');

File: src/assets/download_images.py
import json
import os
import urllib.request

# Get the directory of this script
script_dir = os.path.dirname(os.path.abspath(__file__))

# Load the JSON data
with open(os.path.join(script_dir, 'achievements.json'), 'r') as f:
    data = json.load(f)

# Create the images directory if it doesn't exist
images_dir = os.path.join(script_dir, 'images')
if not os.path.exists(images_dir):
    os.makedirs(images_dir)

# Loop through the achievements and download the images
for achievement in data:
    url = achievement['url']
    identifier = achievement['identifier']
    filename = os.path.join(images_dir, f'{identifier}.png')
    try:
        urllib.request.urlretrieve(url, filename)
    except Exception as e:
        print(f'Warning: Failed to download image for achievement {identifier}: {e}')
